### ISSUE TITLE
Vitest testing

### ISSUE THREAD

**Issue by kevmul** [googlemaps/js-jest-mocks#294]
Thanks for stopping by to let us know something could be better!

---
**PLEASE READ**

If you have a support contract with Google, please create an issue in the [support console](https://cloud.google.com/support/). This will ensure a timely response.

Discover additional support services for the Google Maps Platform, including developer communities, technical guidance, and expert support at the Google Maps Platform [support resources page](https://developers.google.com/maps/support/).

If your bug or feature request is not related to this particular library, please visit the Google Maps Platform [issue trackers](https://developers.google.com/maps/support/#issue_tracker).

Check for answers on StackOverflow with the [google-maps](http://stackoverflow.com/questions/tagged/google-maps) tag.

---

 **Is your feature request related to a problem? Please describe.**
My team and I have used your `Jest Mocks` when we started doing unit tests with `Jest`. But we are switching over to use `Vite` and `Vitest` which has very similar apis. But we cannot use `jest` and `vitest` together.

 **Describe the solution you'd like**
It would be nice to have a package like this targeting the `vitest` testing library (I think just changing over `jest.whatever` to `vi.whater` might be enough?)

 **Describe alternatives you've considered**
I thought about forking this repo for Vitest, but didn't want to maintain such a thing.

 **Additional context**
N/A

**Comment by usefulthink** [comment]
I agree that it would be nice to support compatible testing-libraries as much as possible, but in this case this could be a bit too much.

The only jest-function we actually use is `jest.fn()`, which looks like it has a compatible replacement with `vi.fn()` in vitest.
If those are actually compatible in the long term (idk if it is supposed to stay that way?) we could use an alias instead that uses either `jest.fn` or `vi.fn` based on what is globally available.

Alternatively, it should be possible to inject a simple `window.jest = window.vi` into the vitest-environment, which should also do the trick.

**Comment by tkrotoff** [comment]
Easy to fix:

```TS
import { vi } from "vitest";

globalThis.jest = vi;

... = await import("@googlemaps/jest-mocks");
```

<br>
<br>

(Do not use package [@anshulsanghi/googlemaps-vitest-mocks](https://github.com/anshap1719/js-vitest-mocks) it's a poor solution: no need to fork `@googlemaps/jest-mocks` for 2 simple lines of code + it doesn't seem to be maintained anymore)

**Comment by lucastllima** [comment]
@tkrotoff, @usefulthink  how can I achieve this?
I am using vitest on my project and trying to write your approach:
`
global.jest = vi;
`
but I am receiving `ReferenceError: jest is not defined`

**Comment by usefulthink** [comment]
not sure what `setup-env` is... Is that a file you specified to vitest via the setupFiles option? Also you can `@ts-ignore` those type-errors (or provide the file as js), we know they aren't 100% compatible, but it should be enough for our use-case.

 According to the docs here: https://vitest.dev/config/#setupfiles, it should also be `globalThis` instead of `global`. Beisdes that, I never used vitest, so I can't really tell...

**Comment by robbienohra** [comment]
This is what ended up working for us:

- `setupTests.ts` (top-level)

```ts
vi.stubGlobal('jest', vi)
```

At that point you can actually just proceed to import and use the module as advertised in the original docs

Ie

```ts
import { initialize } from "@googlemaps/jest-mocks";

beforeEach(() => {
  initialize();
});
```
